recaf-ui/src/main/java/me/coley/recaf/ui/pane/SearchPane.java

```diff
@@ -220,12 +210,113 @@ public static SearchPane createDeclarationSearch(String owner, String name, Stri
+			final StringProperty parameters = new SimpleStringProperty();
+
+			enum Type {
+				FIELD, METHOD;
+			}
+
+			final StringProperty owner = new SimpleStringProperty();
+		}
+
+		static class Type {
+			final SetProperty<TypeType> type = new SimpleSetProperty<>();
+
+			enum TypeType {
+				CLASS, RECORD, ENUM, INTERFACE
+			}
+
+		}
+	}
+
+	public static SearchPane creatAdvancedSearch() {
+		StringBinding title = Lang.formatBy("%s: %s", Lang.getBinding("menu.search"),
+			Lang.getBinding("menu.search.advanced"));
+		var advancedSearch = new AdvancedSearch();
+		var whenTargetIsMember = advancedSearch.target.isEqualTo(AdvancedSearch.Target.MEMBER);
+		var whenTargetIsMethod = whenTargetIsMember
+			.and(advancedSearch.member.type.isEqualTo(AdvancedSearch.Member.Type.METHOD));
+		var whenTargetTypeIsDeclaration = advancedSearch.targetType.isEqualTo(AdvancedSearch.TargetType.DECLARATION);
+		var whenTargetIsType = advancedSearch.target.isEqualTo(AdvancedSearch.Target.TYPE);
+
+
+		return advancedSearch.searchPane = new SearchPane(title, Containers
+			.gridAsRows(2)
+			.padding(5, 10, 5, 10)
+			.columnConstraint(c -> c.percentWidth(30))
+			.columnConstraint(c -> c.fillWidth(true).h(HPos.LEFT, Priority.ALWAYS))
+			.vgap(5)
+			.add(label("Target"), choice(AdvancedSearch.Target.values(), AdvancedSearch.Target::name, AdvancedSearch.Target.MEMBER, SelectionMode.SINGLE)
+				.bind(advancedSearch.target))
+			.add(label("Identifier"), textField().bindText(advancedSearch.identifier))
+			.add(label("Deprecated"), checkbox().bind(advancedSearch.deprecated))
+			.add(label("Public"), checkbox().bind(advancedSearch.publiclyAccessible))
+			.addIf(whenTargetIsMember, label("Owner"), textField().bindText(advancedSearch.member.owner))
+			.addIf(whenTargetIsMember,
+				label("Member type"),
+				choice(AdvancedSearch.Member.Type.values(), AdvancedSearch.Member.Type::name, AdvancedSearch.Member.Type.FIELD, SelectionMode.SINGLE)
+					.bind(advancedSearch.member.type)
+			)
+			.addIf(whenTargetIsType,
+				label("Type type"),
+				choice(AdvancedSearch.Type.TypeType.values(), AdvancedSearch.Type.TypeType::name, AdvancedSearch.Type.TypeType.CLASS, SelectionMode.MULTIPLE)
+					.bind(advancedSearch.type.type)
+			)
+			.addIf(whenTargetIsMethod, label("Parameters"), textField().bindText(advancedSearch.member.parameters))
+			.addIf(whenTargetIsMember, nodeSwitch(AdvancedSearch.Member.Type.values(), advancedSearch.member.type)
+					.forCase(AdvancedSearch.Member.Type.FIELD, label("Type"))
+					.forCase(AdvancedSearch.Member.Type.METHOD, label("Return type")),
+				textField().bindText(advancedSearch.member.returnType)
+			)
+			.add(
+				label("Target Type"),
+				choice(AdvancedSearch.TargetType.values(), AdvancedSearch.TargetType::name, AdvancedSearch.TargetType.REFERENCE, SelectionMode.SINGLE)
+					.bind(advancedSearch.targetType)
+			)
+			.addIf(whenTargetTypeIsDeclaration, label("Static"), checkbox().bind(advancedSearch.isStatic))
```

Comment:
Reference search should also allow `static` filtering. Rather than a boolean should this be a 3-option switch? 

Ex:
- dont care
- instance-only
- static-only